```
-- Build files have been written to: /tmp/nix-build-pam_reattach-1.3.drv-0/source/build
cmake: enabled parallel building
building
build flags: -j1 SHELL=/nix/store/1bsjl5incfnszv7scdh4d02sh45vw2w1-bash-5.1-p16/bin/bash
[ 16%] Building C object CMakeFiles/reattach.dir/src/reattach.c.o
In file included from /tmp/nix-build-pam_reattach-1.3.drv-0/source/src/reattach.c:26:
/tmp/nix-build-pam_reattach-1.3.drv-0/source/src/xpc.h:29:10: fatal error: 'xpc/xpc.h' file not found
#include <xpc/xpc.h>
         ^~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/reattach.dir/build.make:76: CMakeFiles/reattach.dir/src/reattach.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/reattach.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/pkgs/os-specific/darwin/pam-reattach/default.nix
+++ b/pkgs/os-specific/darwin/pam-reattach/default.nix
@@ -1,0 +1,45 @@
+{ lib, stdenv, fetchFromGitHub, cmake, openpam, darwin }:
+
+let
+  sdk =
+    if stdenv.isAarch64
+    then null
+    else darwin.apple_sdk.sdk;
+in
+
+stdenv.mkDerivation rec {
+  pname = "pam_reattach";
+  version = "1.3";
+
+  src = fetchFromGitHub {
+    owner = "fabianishere";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1k77kxqszdwgrb50w7algj22pb4fy5b9649cjb08zq9fqrzxcbz7";
+  };
+
+  cmakeFlags = [
+    "-DCMAKE_OSX_ARCHITECTURES=${
+      if stdenv.hostPlatform.system == "x86_64-darwin" then
+        "x86_64"
+      else
+        "arm64"
+    }"
+    "-DENABLE_CLI=ON"
+  ]
+  ++ lib.optional (sdk != null)
+    "-DCMAKE_LIBRARY_PATH=${sdk}/usr/lib";
+
+  buildInputs = [ openpam ]
+    ++ lib.optional (sdk != null) sdk;
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with lib; {
+    homepage = "https://github.com/fabianishere/pam_reattach";
+    description = "Reattach to the user's GUI session on macOS during authentication (for Touch ID support in tmux)";
+    license = licenses.mit;
+    maintainers = with maintainers; [ lockejan ];
+    platforms = platforms.darwin;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2535,6 +2535,8 @@ with pkgs;
 
   plausible = callPackage ../servers/web-apps/plausible { };
 
+  pam-reattach = callPackage ../os-specific/darwin/pam-reattach {};
+
   reattach-to-user-namespace = callPackage ../os-specific/darwin/reattach-to-user-namespace {};
 
   skhd = callPackage ../os-specific/darwin/skhd {

